src/test/java/org/openrewrite/kotlin/tree/FieldAccessTest.java

```diff
@@ -114,4 +121,28 @@ void qualifier() {
           )
         );
     }
+
+    @Test
+    void propertyFieldType() {
+        rewriteRun(
+          spec -> spec.recipe(toRecipe(() -> new KotlinIsoVisitor<>() {
+              @Override
+              public J visitFieldAccess(J.FieldAccess fieldAccess, ExecutionContext ctx) {
+                  if (fieldAccess.getSimpleName().equals("MIN_VALUE")) {
+                      JavaType.Variable fieldType = fieldAccess.getName().getFieldType();
+                      assertThat(fieldType).isNotNull();
+                      assertThat(fieldType.getName()).isEqualTo("MIN_VALUE");
+                      assertThat(fieldType.getType().toString()).isEqualTo("kotlin.Int");
+                      assertThat(fieldType.getOwner().toString()).isEqualTo("kotlin.Int$Companion");
+                  }
+                  return super.visitFieldAccess(fieldAccess, ctx);
+              }
+          })),
+          kotlin(
+            """
+            val i = Int.MIN_VALUE
+            """
+          )
+        );
+    }
```

Comment:
Generally, type mapping tests work through the KotlinTypeGoat, since we can check the type signatures generated in the KotlinTypeSignatureBuilderTest.

I think this is fine for now, I'll add it to KotlinTypeGoat when I have time.